ros/angel_utils/nodes/simple_2d_detection_overlay.cxx

```diff
@@ -406,6 +423,129 @@ ::collect_detections( ObjectDetection2dSet::SharedPtr const det_set )
+    RCLCPP_WARN(
+      log,
+      "Failed to find an image in our history that matches received joints. "
+      "joint source (%zu) < (%zu) history min.",
+      source_nanosec_key, history_min_key
+      );
+    return;
+  }
+
+  // Found something. Double-checking frame source is set to the same thing?
+  if( joints_msg->header.frame_id != find_it->second->header.frame_id )
+  {
+    RCLCPP_WARN(
+      log,
+      "Received joints frame-id does not match the aligned image in "
+      "our history? d-set (%s) != (%s) image",
+      joints_msg->header.frame_id.c_str(),
+      find_it->second->header.frame_id.c_str()
+      );
+    return;
+  }
+
+  // Only plot the patient skeleton
+  if(joints_msg->hand != "patient")
+  {
+    return;
+  }
+
+  // Make a copy of the message image to plot on.
+  cv_bridge::CvImagePtr img_ptr =
+    cv_bridge::toCvCopy( find_it->second, "rgb8" );
+
+  std::map<std::string, std::vector<double>> joint_positions = {};
+  std::map<std::string, std::vector<std::string>> connection_lines = { // mapping of joint line connections
+    {"nose", {"mouth"}},
+    {"mouth", {"throat"}},
+    {"throat", {"chest", "left_upper_arm", "right_upper_arm"}},
+    {"chest", {"back"}},
+    {"left_upper_arm", {"left_lower_arm"}},
+    {"left_lower_arm", {"left_wrist"}},
+    {"left_wrist", {"left_hand"}},
+    {"right_upper_arm", {"right_lower_arm"}},
+    {"right_lower_arm", {"right_wrist"}},
+    {"right_wrist", {"right_hand"}},
+    {"back", {"left_upper_leg", "right_upper_leg"}},
+    {"left_upper_leg", {"left_knee"}},
+    {"left_knee", {"left_lower_leg"}},
+    {"left_lower_leg", {"left_foot"}},
+    {"right_upper_leg", {"right_knee"}},
+    {"right_knee", {"right_lower_leg"}},
+    {"right_lower_leg", {"right_foot"}}
+  };
+
+  static auto const COLOR_PT = cv::Scalar{ 0, 255, 0 }; 
+  int line_thickness = thickness_for_drawing( img_ptr->image );
+  RCLCPP_DEBUG( log, "Using line thickness: %d", line_thickness );
+
+  // Draw the joint points
+  for( auto const& joint : joints_msg->joints )
+  {
+    double x = joint.pose.position.x;
+    double y = joint.pose.position.y;
+    joint_positions[joint.joint] = {x, y}; // save for later
+
+    // Plot the point
+    cv::Point pt = { (int) round( x ),
+                     (int) round( y ) };
+    cv::circle( img_ptr->image, pt, line_thickness*3,
+                COLOR_PT, cv::FILLED);
+  }
+
+  // Draw the joint connections
+  for(auto const& connection : connection_lines)
+  {
+    std::string joint_name = connection.first;
+    std::vector<std::string> joint_connections = connection.second;
+    std::vector<double> first_joint = joint_positions[joint_name];
+    cv::Point pt1 = { 
+      (int) round( first_joint[0] ),
+      (int) round( first_joint[1] ) 
+    };
+
+    for(auto const& connecting_joint : joint_connections)
+    {
+      std::vector<double> connecting_pt = joint_positions[connecting_joint];
+      cv::Point pt2 = { 
```

Comment:
We could probably define `pt2` outside of this for-loop, just under `pt1`'s definition, and then assign to the variable inside this nested for-loop. The compiler may be optimizing this for us, but constantly re-allocating the memory for a type is a non-zero cost that should be easy to eliminate here.